### PR TITLE
CEC-821

### PR DESCRIPTION
**Pull request by poojantcs**

### CODE REVIEW

File: src/e2e-test/features/successfactors/DesignTime.feature [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+# the License.
+
+
+@SuccessFactorsSource
+@Smoke
+@Regression
+Feature: SuccessFactors Source - Design time scenarios
+
+  @BATCH-TS-SCFA-DSGN-01
+  Scenario Outline: Verify user should be able to get output schema for the plugin
```
Comment (poojantcs) [inline src/e2e-test/features/successfactors/DesignTime.feature]: Re-word: ".. to get output schema for a valid Entity name"

File: src/e2e-test/features/successfactors/DesignTime.feature [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    When Open Datafusion Project to configure pipeline
+    And Select data pipeline type as: "Batch"
+    And Select plugin: "SAP SuccessFactors" from the plugins list as: "Source"
+    And Navigate to the properties page of plugin: "SAP SuccessFactors"
+    And Enter input plugin property: "referenceName" with value: "Referencename"
+    And Enter input plugin property: "baseURL" with value: "admin.baseurl"
+    And Enter input plugin property: "entityName" with value: "<EntityName>"
+    And Enter input plugin property: "username" with value: "admin.username"
+    And Enter input plugin property: "password" with value: "admin.password"
+    And Validate output schema with expectedSchema "<ExpectedSchema>"
```
Comment (poojantcs) [inline src/e2e-test/features/successfactors/DesignTime.feature]: Use step with the loggers here: "Verify the Output Schema matches the Expected Schema: {string}"

Comment (poojantcs) [reply src/e2e-test/features/successfactors/DesignTime.feature]: Also, we need to enhance this step for hierarchical /record type fields. Logged a ticket: https://cloudsufi.atlassian.net/browse/CEC-851
We will discuss the approach.

File: src/e2e-test/features/successfactors/DesignTime.feature [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    Examples: 
+    | EntityName             | ExpectedSchema                |
+    | BadgeTemplates         | schema.badgetemplates         |
+    | AdvancesAccumulation   | schema.advancesaccumulation   |
+    | Advance                | schema.advance                |
+    | EmpCompensation        | schema.empcompensation        |
+    | Background_Certificates| schema.backgroundcertificates |
+
+  @BATCH-TS-SCFA-DSGN-02
+  Scenario Outline: Verify user should be able to validate the plugin for Filter options property
```
Comment (poojantcs) [inline src/e2e-test/features/successfactors/DesignTime.feature]: Re-word: "validate the plugins **with** Filter options property"

Comment (poojantcs) [reply src/e2e-test/features/successfactors/DesignTime.feature]: Apply the same change in other scenarios of this feature.

File: src/e2e-test/features/successfactors/DesignTimeValidation.feature [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    Then Verify mandatory property error for below listed properties:
+      | referenceName    |
+      | baseURL          |
+      | entityName       |
+      | username         |
+      | password         |
+
+
+  @BATCH-TS-SCFA-DSGN-ERROR-02
+  Scenario: Verify validation message when user provide invalid BaseURL
```
Comment (poojantcs) [inline src/e2e-test/features/successfactors/DesignTimeValidation.feature]: *provides an

Comment (poojantcs) [reply src/e2e-test/features/successfactors/DesignTimeValidation.feature]: Apply similar changes in other places as well

File: src/e2e-test/features/successfactors/RunTime.feature [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+    And Validate "BigQuery" plugin properties
+    And Close the Plugin Properties page
+    And Connect source as "SAP-SuccessFactors" and sink as "BigQueryTable" to establish connection
+    And Save the pipeline
+    And Preview and run the pipeline
+    And Verify the preview of pipeline is "success"
+    Then Verify sink plugin's Preview Data for Input Records table and the Input Schema matches the Output Schema of Source plugin
+
+  @BATCH-TS-SCFA-DSGN-RNTM-02 @BQ_SINK_TEST
+  Scenario: Verify user should be able to run and deploy the pipeline when plugin is configured for valid Entity name
```
Comment (poojantcs) [inline src/e2e-test/features/successfactors/RunTime.feature]: "should be able to deploy and run the pipeline"

File: src/e2e-test/java/io/cdap/plugin/tests/runner/TestRunner.java [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+import io.cucumber.junit.CucumberOptions;
+import org.junit.runner.RunWith;
+
+/**
+ * Test Runner to execute SuccessFactors plugin's test scenarios.
+ */
+@RunWith(Cucumber.class)
+@CucumberOptions(
+  features = {"src/e2e-test/features"},
+  glue = {"io.cdap.plugin.successfactorssource.stepsdesign",
```
Comment (poojantcs) [inline src/e2e-test/java/io/cdap/plugin/tests/runner/TestRunner.java]: Update this with correct paths

**Comment by poojantcs** [comment]
Need to add Run time validation tests. Ticket logged: https://cloudsufi.atlassian.net/browse/CEC-852

File: src/e2e-test/features/successfactors/DesignTime.feature (hunk 1 above)
Comment (Vipinofficial11) [reply src/e2e-test/features/successfactors/DesignTime.feature]: Updated

File: src/e2e-test/features/successfactors/DesignTime.feature (hunk 2 above)
Comment (Vipinofficial11) [reply src/e2e-test/features/successfactors/DesignTime.feature]: Verify the Output Schema matches the Expected Schema: {string} -> This step actually doesn't click on the get schema button and we have the get schema button in this plugin. Also, the function which is checking the types of fields is same in both the steps so this one also has logger.

File: src/e2e-test/features/successfactors/DesignTime.feature (hunk 3 above)
Comment (Vipinofficial11) [reply src/e2e-test/features/successfactors/DesignTime.feature]: Updated

File: src/e2e-test/features/successfactors/DesignTimeValidation.feature (hunk 4 above)
Comment (Vipinofficial11) [reply src/e2e-test/features/successfactors/DesignTimeValidation.feature]: updated

File: src/e2e-test/features/successfactors/RunTime.feature (hunk 5 above)
Comment (Vipinofficial11) [reply src/e2e-test/features/successfactors/RunTime.feature]: updated

**Pull request closed without merging**